runelite-client/src/main/java/net/runelite/client/plugins/driftnet/DriftNetPlugin.java

```diff
@@ -0,0 +1,258 @@
+		{
+			northNet = new DriftNet();
+			southNet = new DriftNet();
+			interacting = null;
+			fishes.clear();
+		}
+	}
+
+	@Provides
+	DriftNetConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DriftNetConfig.class);
+	}
+
+	@Subscribe
+	public void onGameObjectSpawned(GameObjectSpawned event)
+	{
+		GameObject gameObject = event.getGameObject();
+		if (event.getGameObject().getId() == SOUTH_NET_ID)
+		{
+			southNet.setDriftNet(gameObject);
+		}
+		else if (event.getGameObject().getId() == NORTH_NET_ID)
+		{
+			northNet.setDriftNet(gameObject);
+		}
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		DriftNet net = null;
+		if (event.getIdentifier() == SOUTH_NET_ID)
+		{
+			net = southNet;
+		}
+		else if (event.getIdentifier() == NORTH_NET_ID)
+		{
+			net = northNet;
+		}
+		if (net != null && event.getOption() != null)
+		{
+			if (event.getTarget().toLowerCase().contains("(full)"))
+			{
+				net.setNetStatus(DriftNet.DriftNetStatus.FULL);
+			}
+			else if (event.getOption().toLowerCase().contains("set"))
+			{
+				net.setNetStatus(DriftNet.DriftNetStatus.UNSET);
+			}
+			else
+			{
+				net.setNetStatus(DriftNet.DriftNetStatus.SET);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameObjectDespawned(GameObjectDespawned event)
+	{
+		if (event.getGameObject().getId() == SOUTH_NET_ID)
+		{
+			southNet.setDriftNet(null);
+		}
+		else if (event.getGameObject().getId() == NORTH_NET_ID)
+		{
+			northNet.setDriftNet(null);
+		}
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		NPC npc = event.getNpc();
+		if (npc.getId() == NpcID.FISH_SHOAL)
+		{
+			fishes.put(npc, -1L);
+		}
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned event)
+	{
+		NPC npc = event.getNpc();
+		if (npc.getId() == NpcID.FISH_SHOAL)
+		{
+			fishes.remove(npc);
+		}
+	}
+
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged event)
+	{
+		if (event.getSource() == client.getLocalPlayer()
+			&& event.getTarget() instanceof NPC
+			&& ((NPC) event.getTarget()).getId() == NpcID.FISH_SHOAL)
+		{
+			interacting = (NPC) event.getTarget();
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick tick)
+	{
+		for (NPC fish : fishes.keySet())
+		{
+			if (fishes.get(fish) > 0
+				&& System.currentTimeMillis() - fishes.get(fish) > 1000 * config.highlightDuration())
+			{
+				fishes.put(fish, -1L);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() != ChatMessageType.SPAM && event.getType() != ChatMessageType.GAMEMESSAGE)
+		{
+			return;
+		}
+		if (event.getMessage().toLowerCase().contains("prod at"))
+		{
+			fishes.put(interacting, System.currentTimeMillis());
+		}
+	}
+
+	private int supNorm(WorldPoint w1, WorldPoint w2)
+	{
+		int x = abs(w1.getX() - w2.getX());
+		int y = abs(w1.getY() - w2.getY());
+		return Math.max(x, y);
+	}
+
+	boolean isInDriftNetArea()
```

Comment:
If you define a `WorldArea` for the fish area, you can check if the distance to that world area is 0, instead of using the fish.

Reply:
Hi, would this work for an instanced area? The drift net fishing is instanced.

Reply:
Have a play around with it, it should be possible

Reply:
How do I get the coordinates for instanced areas? It changes every time.

Reply:
How to get coordinates for instanced areas? It changes every time.